lib/puppet/face/epp.rb

```diff
@@ -104,13 +104,19 @@
 
   action (:dump) do
     summary _("Outputs a dump of the internal template parse tree for debugging")
-    arguments "-e <source> | [<templates> ...] "
+    arguments "[--format <old|pn|json>] [--pretty] { -e <source> | [<templates> ...] } "
     returns _("A dump of the resulting AST model unless there are syntax or validation errors.")
     description <<-'EOT'
       The dump action parses and validates the EPP syntax and dumps the resulting AST model
       in a human readable (but not necessarily an easy to understand) format.
-      The output format of the dumped tree is intended for epp parser debugging purposes
-      and is not API, and may thus change between versions without deprecation warnings.
+
+      The output format can be controlled using the --format <old|pn|json>
+      'old' is the default, but now deprecated format which is not API.
+      'pn' is the Puppet Extended S-Expression Notation.
+      'json' outputs the same graph as 'pn' but with JSON syntax.
```

Comment:
Maybe line 113 should end with `where:` and each of the three formats starts with a `*` bullet ?